support/k8s/k8s-workload-registrar/config_crd.go

```diff
@@ -51,6 +55,41 @@ func (c *CRDMode) ParseConfig(hclConfig string) error {
 		c.WebhookPort = defaultWebhookPort
 	}
 
+	if c.IdentityTemplate != "" && c.CommonMode.PodAnnotation != "" {
+		return errs.New("workload registration configuration is incorrect, can't specify both identity_template and pod_annotation")
+	}
+
+	if c.IdentityTemplate != "" && c.CommonMode.PodLabel != "" {
+		return errs.New("workload registration configuration is incorrect, can't specify both identity_template and pod_label")
+	}
+
+	// eliminate orphaned identity_template_label
+	if c.IdentityTemplateLabel != "" && c.IdentityTemplate == "" {
+		return errs.New("identity_template_label defined without identity_template")
+	}
+
+	// eliminate orphaned context
+	if c.Context != nil && c.IdentityTemplate == "" {
+		return errs.New("context defined without identity_template")
+	}
+
+	// eliminate reference to non-existing context
+	if c.Context == nil && c.IdentityTemplate != "" && strings.Contains(c.IdentityTemplate, "{{.Context.") {
+		return errs.New("identity_template references non-existing context")
+	}
+
+	// IdentityTemplate represents the format following the trust domain and as such, it must not begin with spiffe://, // or /
+	if strings.HasPrefix(c.IdentityTemplate, "spiffe://") ||
+		strings.HasPrefix(c.IdentityTemplate, "/") {
+		return errs.New("identity_template cannot start with spiffe:// or /")
+	}
+
+	if c.CommonMode.Mode == modeCRD {
+		if c.IdentityTemplate == "" && c.CommonMode.PodAnnotation == "" && c.CommonMode.PodLabel == "" {
+			return errs.New("in crd mode, only one workload registration mode is allowed: identity_template, pod_annotation or pod_label")
+		}
+	}
```

Comment:
I think this check might be covered by the one i made a suggestion on above.

Reply:
Actually, this one makes sure at least one of the three is selected. I changed a bit the error message to make it less ambiguous.

Reply:
i see now. I think we want to preserve the service account based registration for backwards compatibility.

Reply:
we can do that by setting the default value of IdentityTemplate to match what the service account based registration would do.

Reply:
I agree with you @faisal-memon . What do you think if we assign the default, service account-based IdentityTemplate in Config_crd, when none of the other options is set (label, annotation) and/or when `identity_template=""`? 

Typically, `identity_template_label` would probably be missing as well, so we don't risk creating SPIFFEID for every pod in the cluster. All pods get SPIFFEID only if `identity_template_label=""`

Reply:
sounds good to me.